### PR TITLE
feat(design): #18 Designed the login page

### PR DESCRIPTION
**Pull request by VenkatVenkatesh**
1. Used `mat-buttons` for creating buttons.
2. Added `FlexLayout module` for using flexbox.
3. Created a styles folder and moved the SASS variables to `variables.scss` file.
4. Added social svg icons to the assets folder and registered them using `mat-icon-registry`
Screen Shot 2019-04-10 at 1 40 16 PM

### CODE REVIEW

**Review by SrijithRad — DISMISSED** [review]

File: web/src/app/app.material-module.ts [hunk 1]
```diff
@@ -2,14 +2,38 @@ import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
...
   ],
   providers: [],
   bootstrap: [],
 })
-export class AppMaterialModule { }
+export class AppMaterialModule {
+  constructor(private matIconRegistry: MatIconRegistry, private domSanitizer: DomSanitizer) {
+    this.matIconRegistry.addSvgIcon(
+      'github',
+      this.domSanitizer.bypassSecurityTrustResourceUrl('../assets/images/Github.svg')
```
Comment (eddiejaoude) [inline web/src/app/app.material-module.ts]: the image names are inconsistent with some capitalised and some not. Can we make them all lowercase please

File: web/src/app/app-routing.module.ts [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
+const routes: Routes = [
+  { path: 'login', component: LoginPageComponent }
+];
 
 @NgModule({
   imports: [RouterModule.forRoot(routes)],
   exports: [RouterModule]
 })
 export class AppRoutingModule { }
+export const routingComponent = [LoginPageComponent];
```
Comment (eddiejaoude) [inline web/src/app/app-routing.module.ts]: Can this be consistent and be capitalised

File: web/src/app/app.material-module.ts (hunk 1 above)
Comment (VenkatVenkatesh) [reply web/src/app/app.material-module.ts]: Okay @eddiejaoude . Will do it 👍

File: web/src/app/components/login-page/login-page.component.html [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<div class="card-container">
+  <mat-card class="card" fxFlex="30">
+    <mat-card-content fxLayout="column" fxLayoutAlign="space-between space-between" fxLayoutGap="20px">
+      <mat-card-title fxLayoutAlign="center">Sign into Dashboard</mat-card-title>
+      <button mat-flat-button class="github" >
```
Comment (eddiejaoude) [inline web/src/app/components/login-page/login-page.component.html]: > `class="github" >`
extra space not needed after the class

`class="github">`

File: web/src/app/components/login-page/login-page.component.html [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        <mat-icon svgIcon="twitter"></mat-icon>
+        <span>Sign in with twitter</span>
+        </button>
+      <button mat-flat-button class="instagram">
+        <mat-icon svgIcon="instagram"></mat-icon>
+        <span>Sign in with Instagram</span>
+      </button>
+    </mat-card-content>
+  </mat-card>
+</div>
```
Comment (eddiejaoude) [inline web/src/app/components/login-page/login-page.component.html]: Missing blank line at the end of the file. The IDE should be add this automatically

File: web/src/app/components/login-page/login-page.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  beforeEach(() => {
+    fixture = TestBed.createComponent(LoginPageComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+});
```
Comment (eddiejaoude) [inline web/src/app/components/login-page/login-page.component.spec.ts]: As we are not using these test at the moment, probably best to remove these files

File: web/src/styles/variables.scss [hunk 6]
```diff
@@ -0,0 +1,10 @@
+//colors
+$github-primary: black;
+$twitter-primary: #1da1f2;
+$youtube-primary: #ff0000;
+$instagram-primary: rgba(248, 248, 248, 0.3);
+$text-color-primary: black;
+$text-color-accent: white;
+
+//heights
+$button-height: 45px;
```
Comment (eddiejaoude) [inline web/src/styles/variables.scss]: Can we use relative sizing in most places, where it makes sense to

File: web/src/styles/variables.scss [hunk 7]
```diff
@@ -0,0 +1,10 @@
+//colors
+$github-primary: black;
+$twitter-primary: #1da1f2;
+$youtube-primary: #ff0000;
+$instagram-primary: rgba(248, 248, 248, 0.3);
+$text-color-primary: black;
+$text-color-accent: white;
```
Comment (eddiejaoude) [inline web/src/styles/variables.scss]: Can the colours be consistent format, thanks

**Comment by eddiejaoude** [comment]
Had a quick glance and left a few comments, sorry for the short messages 🤓

**Comment by VenkatVenkatesh** [comment]
Will address the review comments and let you know Eddie.

**Comment by VenkatVenkatesh** [comment]
Hi Eddie, I have worked on the review comments. Could you please check and let me know if you have any issues.

**Review by eddiejaoude — APPROVED** [review]

**Comment by eddiejaoude** [comment]
Looks good, thanks 👍

**Comment by VenkatVenkatesh** [comment]
Thank you :+1:

**Pull request merged**